_posts/2022-03-25-unnormalized-resnets.md

```diff
@@ -0,0 +1,725 @@
+
+where $\mu_g = \sum_{\boldsymbol{x} \in \mathcal{B}} \nabla_{\hat{\boldsymbol{x}}} \mathcal{L}$ and $\operatorname{cov}(\boldsymbol{g}, \hat{\boldsymbol{x}}) = \frac{1}{|\mathcal{B} |} \sum_{\boldsymbol{x} \in \mathcal{B}} \boldsymbol{g} \odot \hat{\boldsymbol{x}}.$
+Note that this directly corresponds to centering the gradients, which is also supposed to improve learning speed ([Schraudolph, 1998](#schraudolph98centering)).
+
+In the end, everyone seems to agree that one of the main benefits of BN is that it enables higher learning rates ([Ioffe & Szegedy, 2015](#ioffe15batchnorm); [Bjorck et al., 2018](#bjorck18understanding); [Santurkar et al., 2018](#santurkar18how); [Luo et al., 2019](#luo19towards)), which results in faster learning and better generalization.
+An additional benefit is that BN is scale-invariant and therefore much less sensitive to weight initialization ([Ioffe & Szegedy, 2015](#ioffe15batchnorm); [Ioffe, 2017](#ioffe17batchrenorm)).
+
+### Alternatives
+
+Why would we ever want to get rid of BN then?
+Although BN provides important benefits, it also comes with a few downsides:
+
+ - BN does not work well with **small batch sizes** ([Ba et al., 2016](#ba16layernorm); [Salimans & Kingma, 2016](#salimans16weightnorm); [Ioffe, 2017](#ioffe17batchrenorm)).
+   For a batch size of one, we have zero standard deviation, but also with a few samples, the estimated statistics are often not accurate enough.
+ - BN is not directly applicable to certain input types ([Ba et al. 2016](#ba16layernorm); also see Figure&nbsp;[1](#fig_dims)) and performs poorly when there are **dependencies between samples** in a mini-batch ([Ioffe, 2017](#ioffe17batchrenorm)).
+ - BN uses **different statistics for inference** than those used during training ([Ba et al., 2016](#ba16layernorm); [Ioffe, 2017](#ioffe17batchrenorm)).
+   This is especially problematic if the distribution during inference is different or drifts away from the training distribution.
+ - BN does not play well with **other regularization** methods ([Hoffer et al., 2018](#hoffer18norm)).
+   This is especially known for $\mathrm{L}_2$-regularization ([Hoffer et al., 2018](#hoffer18norm)) and dropout ([Li et al., 2019](#li19understanding)).
+ - BN introduces a significant **computational overhead** during training ([Ba et al., 2016](#ba16layernorm); [Salimans & Kingma, 2016](#salimans16weightnorm); [Gitman and Ginsburg, 2017](#gitman17comparison)).
+   Because of the running averages, also memory requirements increase when introducing BN.
+
+Therefore, alternative normalization methods have been proposed to solve one or more of the problems listed above while trying to maintain the benefits of BN.
+
+<figure id="fig_dims">
+    <img src="{{ site.url }}/public/images/2021-12-01-unnormalized-resnets/data_dimensions.svg" alt="visualization of different input data types">
```

Comment:
You'll need to update the image paths to the new directory i.e. `2021-12-01` -> `2022-03-25` throughout all of the post (sorry about that)